doc/modules/grid_search.rst

```diff
@@ -167,6 +171,339 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
+candidates at the last iteration. On the other hand if we start with a small
+number of candidates, the last iteration might use less than 640 samples
+which is a waste of resources.
+
+In the case of :class:`HalvingGridSearchCV`, the number of candidates is set
+by default such that the maximum amount of resources is used at the last
+iteration.
+
+Changing the value of ``min_resources`` will impact the number of possible
+iterations, and as a result will also have an effect on the ideal number of
+candidates.
+
+Another consideration when choosing ``min_resources`` is whether or not it
+is easy to discriminate between good and bad candidates with a small amount
+of resources. For example, if you need a lot of samples to distinguish
+between good and bad parameters, a high ``min_resources`` (possibly with the
+use of ``aggressive_elimination=True``) is recommended. On the other hand if
+the distinction is clear even with a small amount of samples, then a small
+``min_resources`` may be preferable since it would speed up the computation.
+
+By default, ``min_resources`` is set to a small value (see docstrings for
+details) that depends on the number of folds, and the number of classes for
+classification problems. Depending on the setting, the default value of
+``min_resources`` might not be ideal.
+
+.. note::
+  Notice in the example above that the last iteration does not use the
+  maximum amount of resources available: 1000 samples are available, yet
+  only 640 are used. Using ``force_exhaust_resources=True`` will set
+  ``min_resources`` to a specific value such that the last iteration uses as
+  many samples as possible. Please see :ref:`exhausting_the_resources` for
+  details.
+
+.. _amount_of_resource_and_number_of_candidates:
+
+Amount of resource and number of candidates at each iteration
+-------------------------------------------------------------
+
+The amount of resources ``resource_iter`` (e.g. the number of samples)
+allocated for each candidate at iteration ``i`` is controlled by the
+parameters ``ratio`` and ``min_resources`` as follows::
+
+    resource_iter = ratio**i * min_resources
+
+``min_resources`` is the amount of resources used at the first iteration and
+``ratio`` defines the proportions of candidates that will be selected for
+the next iteration (``ratio`` must be greater than 1)::
+
+    n_candidates_to_keep = n_candidates_at_i // ratio
+
+So in the first iteration, we use ``min_resources`` resources
+``n_candidates`` times. In the second iteration, we use ``min_resources *
+ratio`` resources ``n_candidates // ratio`` times. The third again
+multiplies the resources per candidate and divides the number of candidates.
+This process stops when the maximum amount of resource per candidate is
+reached, or when we have identified the best candidate. The best candidate
+is identified at the iteration that is evaluating `ratio` or less candidates
+(see below).
```

Comment:
Thanks a lot for your review @adrinjalali .

I think I addressed most points. I also updated here the UG about `ratio`. I hope it's clearer now why having `ratio` or less candidates at the last iteration is what we want. LMK!